### ISSUE TITLE
[proposal] The fate of visualmetrics.py

### ISSUE THREAD

**Issue by tarekziade** [sitespeedio/browsertime#1142]
We're currently modifying visualmetrics.py enough from its original version that I think we should make it an official part of the browsertime code and not a vendored thing anymore.

To make our life easier in testing and maintaining it, I am proposing these changes:

- [x] create a "browsertime" python package in the repository and move visualmetrics.py and test_visualmetrics.py in it
- [x] make visualmetrics.py an installable script
- [x] make it a project by adding a pyproject.toml file
- [x] add Tox support so we can run lints and tests on Python 2.7 and 3.8 (with coverage)
- [x] format it using Black
- [x] add coverage information

In follow up work:

- [ ] add a version (right now it's "0.1")
- [ ] add tests until we reach 80% coverage of the module, and make it Py 3 compatible along the way

The impact for people using visualmetrics.py will be:
- it would move from vendor/ to browsertime/ directory in the repo
- it will become an installable script ("pip install browsertime")
- maybe it would make sense to rename it to avoid any confusion in the future

**Comment by soulgalore** [comment]
Sounds great @tarekziade ! Lets just iterate over the naming :)

**Comment by tarekziade** [comment]
work in progress in https://github.com/sitespeedio/browsertime/tree/tox

**Comment by ncalexan** [comment]
I am 100% in favour of improving the existing Python script.  I'd very much like it to do its work in parallel, for example, and potentially not invoke IM to do what -- I think -- are standard image transformations.

**Comment by soulgalore** [comment]
@tarekziade a side note: we also use https://github.com/paulirish/speedline to calculate speed index from the Chrome devtools log (as an alternative using Visual Metrics).

I think for Firefox using the internalWindowrecorder we could rework Speedline to calculate the same thing but for Firefox (or plain images with timestamps), that way getting Visual Metrics for Firefox would work without python/imagemagick and ffmpeg?

**Comment by tarekziade** [comment]
That could make deployments easier indeed - does this tool covers the same metrics ?

**Comment by soulgalore** [comment]
We will miss out Contentful Speed Index. It support First Visual Change, Speed Index, Perceptual Speed Index and Last Visual Change. That lib was done by https://github.com/pmdartus a long time ago and then later adopted by the Chrome team and used by Lighthouse. I'm thinking that the core functionality we need is there and we could either ask if they accept PRs to make it more generic or give credit and redo our own modified version?

**Comment by soulgalore** [comment]
There's a couple of things I would like to fix before we close this issue:
* I've moved testing to GitHub Actions but I don't get the Visual Metrics tests to to work. If you @tarekziade could help me with that it would be great: https://github.com/sitespeedio/browsertime/issues/1303
* I think the [browsertime folder](https://github.com/sitespeedio/browsertime/tree/main/browsertime) name for Visual Metrics is confusing. Shouldn't be named something with Visual Metrics?
* Running with Python 3 I don't get Visual Metrics log file (so I can debug), I think something maybe is wrong with the setup. I've added an error log so I remember but if you also could try @tarekziade and see if you get that too, that would be great. `[2020-09-23 13:16:40] ERROR: Could not remove the log file from VisualMetrics, does it exist?: /.../browsertime/browsertime-results/en.m.wikipedia.org-wiki-Barack_Obama/2020-09-23T131553+0200/pages/en.m.wikipedia.org/wiki/Barack_Obama/data/video/1.visualmetrics.log`
